### PR TITLE
[User] #419 Generate API keys when API access is enabled

### PR DESCRIPTION
**Pull request by crouillon**
- Generate API keys when need
- Add User() unit tests

### CODE REVIEW

File: Security/Tests/UserTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        $this->assertEquals('firstname lastname (login)', $this->mockUser->__toString());
+    }
+
+    /**
+     * @covers ::serialize
+     */
+    public function testSerialize()
+    {
+        $this->assertEquals('{"username":"login","commonname":"firstname lastname"}', $this->mockUser->serialize());
```
Comment (eric-chau) [inline Security/Tests/UserTest.php]: It is kinda confusing to call ``serialize`` but the result seems to be json format.

File: Security/Tests/UserTest.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     * @covers ::generateRandomApiKey
+     * @covers ::generateApiPublicKey
+     */
+    public function testGenerateRandomApiKey()
+    {
+        $this->mockUser->generateRandomApiKey();
+        $this->assertNotNull($this->mockUser->getApiKeyPrivate());
+        $this->assertEquals(
+                sha1($this->mockUser->getCreated()->format(\DateTime::ATOM) . $this->mockUser->getApiKeyPrivate()), $this->mockUser->getApiKeyPublic()
+        );
```
Comment (eric-chau) [inline Security/Tests/UserTest.php]: Please don't write line with more that 120 characters =)

```php
$this->assertEquals(
    sha1($this->mockUser->getCreated()->format(\DateTime::ATOM) . $this->mockUser->getApiKeyPrivate()), 
    $this->mockUser->getApiKeyPublic()
);
```

**Comment by eric-chau** [comment]
:+1: nice contribution @crouillon, tell me when you want to merge it.

**Comment by mickaelandrieu** [comment]
:+1:

**Pull request merged**